test/markup/latex/control_sequences.expect.txt

```diff
@@ -0,0 +1,34 @@
+<span class="hljs-keyword">\foo</span> bar
+<span class="hljs-keyword">\foo</span>:bar
+<span class="hljs-keyword">\ </span>and<span class="hljs-keyword">\
+</span>bar<span class="hljs-keyword">\foo</span>:baz
+<span class="hljs-keyword">\!</span>!
+<span class="hljs-keyword">\^^3a</span>bc
+<span class="hljs-keyword">\^^3</span>ABC
+<span class="hljs-keyword">\^^x</span>yz
+<span class="hljs-keyword">\^</span><span class="hljs-built_in">^</span>°!
+
+<span class="hljs-comment">% @ is always treated as a letter</span>
+<span class="hljs-keyword">\@foo</span>
+<span class="hljs-keyword">\foo@bar</span>
+<span class="hljs-keyword">\@ifnextchar</span>[<span class="hljs-keyword">\@foo</span><span class="hljs-keyword">\@@foo</span>
+
+<span class="hljs-comment">% expl3 macros are recognized</span>
+<span class="hljs-keyword">\cs_new:Nn</span> <span class="hljs-keyword">\hljs_test:N</span> <span class="hljs-keyword">\__hljs_internal:Nx</span>
+<span class="hljs-keyword">\use_ii_i:nn</span>
+<span class="hljs-keyword">\cs:w</span> hljs<span class="hljs-built_in">_</span>test:N<span class="hljs-keyword">\cs_end:</span>
+<span class="hljs-keyword">\use_i:nn</span> <span class="hljs-keyword">\use:</span><span class="hljs-keyword">\q_test</span>
+<span class="hljs-keyword">\if:w</span><span class="hljs-keyword">\scan_stop:</span><span class="hljs-keyword">\s_test</span>[<span class="hljs-keyword">\else:</span>]<span class="hljs-keyword">\fi:</span>
+<span class="hljs-keyword">\hbox:n</span> <span class="hljs-keyword">\vbox:n</span>
+<span class="hljs-keyword">\::c</span><span class="hljs-keyword">\::o</span><span class="hljs-keyword">\::V</span><span class="hljs-keyword">\:::</span><span class="hljs-keyword">\hljs_test_args:Nnn</span>
+<span class="hljs-keyword">\::V_unbraced</span>
+<span class="hljs-keyword">\l_hljs_var_tl</span> <span class="hljs-keyword">\g_tmpa_prop</span> <span class="hljs-keyword">\c__hljs_secret_int</span>
+<span class="hljs-keyword">\c_e_fp</span>
+<span class="hljs-comment">% but only those conforming to the naming scheme</span>
+<span class="hljs-comment">% with at least to letters in every name</span>
```

Comment:
```suggestion
<span class="hljs-comment">% with at least two letters in every name</span>
```
You meant two right?  Is this still accurate?

Reply:
Are there examples here that should and shouldn't be highlighted?  I'm trying to make heads or tails of it but it's hard. :)

Reply:
Yes. The lines above this comment should be highlighted as LaTeX3 macros, the ones below shouldn't.

The naming convention is as follows
- A function belongs to a `module`, has a `function name` and a `signature`. These are encoded in its macro name as `\module_function_name:signature` or `\__module_function_name:signature`.
- A variable has a `scope`, a `module`, a `variable_name` and a `type`. These are encoded in its macro name as `\scope_module_variable_name_type` or `\scope__module_variable_name_type`. The module may be omitted if it is identical to the type.
- Quarks have a `quark name` and look like `\q_quark_name`.
- Similarly, scan tokens look like `\s_scan_token_name`.
- Built-in functions often do not follow any scheme except for the signature: `\anything:signature`

Now, the rule detecting LaTeX3 macros in the grammar only detects macro names that have at least two characters in each of `module`, `function_name`, `type`, `quark_name` and `scan_mark_name`, as applicable (except for certain built-in functions). (`scope` is always one of `l`, `g` or `c`, there is not restriction on `variable_name`s. Double `_`, except where indicated above, are also prohibited.) The restriction on at least two characters was introduced to avoid things like the misinterpretation of `\theta_a` (which is the letter theta with the subscript a in math mode) as a LaTeX3 variable.

That's what I wanted to hint at in that comment.

Reply:
A few more comments here maybe wouldn't hurt for non-latex people, but I won't say they are necessary as I really want to get this merged, not keeping giving you more work to do. :-)  Or perhaps it's a lost cause for the non-latex among us. :-)

Reply:
Yeah, I thought about that, but wanted to wait until I was sure the rule wouldn't change anymore. What would be the best place for this? The grammar file or the test file? Or both?

Reply:
Definitely the grammar though you could add a `(see grammar for details)` hint or something in the test I think.

Reply:
Ok, give me a moment, I will add some comments.